src/libsyncengine/propagation/executor/executorworker.cpp

```diff
@@ -679,25 +621,20 @@ bool ExecutorWorker::generateCreateJob(SyncOpPtr syncOp, std::shared_ptr<Abstrac
             } catch (std::exception const &e) {
                 LOGW_SYNCPAL_WARN(_logger, L"Error in CreateDirJob::CreateDirJob for driveDbId="
                                                    << _syncPal->driveDbId() << L" : " << Utility::s2ws(e.what()).c_str());
-                _executorExitCode = ExitCode::DataError;
-                _executorExitCause = ExitCause::Unknown;
-                return false;
+                return ExitCode::DataError;
             }
         } else {
-            _executorExitCode = convertToPlaceholder(relativeLocalFilePath, true, _executorExitCause);
-            if (_executorExitCode != ExitCode::Ok) {
+            if (ExitInfo exitInfo = convertToPlaceholder(relativeLocalFilePath, true); !exitInfo) {
                 LOGW_SYNCPAL_WARN(_logger, L"Failed to convert to placeholder for: "
                                                    << SyncName2WStr(syncOp->affectedNode()->name()).c_str());
-                return false;
+                return exitInfo;
             }
 
             uint64_t filesize = 0;
-            if (!getFileSize(absoluteLocalFilePath, filesize)) {
+            if (ExitInfo exitInfo = getFileSize(absoluteLocalFilePath, filesize); !exitInfo) {
```

Comment:
Same remark as for checkIfPathExists.

Reply:
I'm not sure I understand the link between those two comments.
(Assuming you are referring to [this discussion](https://github.com/Infomaniak/desktop-kDrive/pull/356#discussion_r1822019181).)